### PR TITLE
Add extensions for setting soft input mode in Fragments

### PR DESCRIPTION
**Pull request by zsmb13**
### Description

We need different window handling behaviour depending on which Fragment we're in. Sometimes we want to resize the window so that it all fits above the keyboard, and sometimes we want to cover its bottom half with the keyboard.

This PR removes adds extensions for `Fragment` to dynamically set this for each screen.

https://user-images.githubusercontent.com/12054216/102590365-beb8db00-4110-11eb-9626-51437c4ffbed.mov

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] ~Changelog updated with client-facing changes~
- [ ] ~New code is covered by unit tests~
- [ ] ~Comparison screenshots added for visual changes~
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/channel/add/AddChannelFragment.kt [hunk 1]
```diff
@@ -38,6 +39,11 @@ class AddChannelFragment : Fragment() {
         bindAddChannelView()
     }
 
+    override fun onResume() {
+        super.onResume()
+        useAdjustResize()
```
Comment (bychkovdmitry) [inline stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/channel/add/AddChannelFragment.kt:44]: Can we remove android:windowSoftInputMode="adjustResize" from the manifest now?

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/util/extensions/Fragment.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import androidx.fragment.app.Fragment
+
+fun Fragment.useAdjustNothing() = setSoftInputMode(SOFT_INPUT_ADJUST_NOTHING)
+
+fun Fragment.useAdjustPan() = setSoftInputMode(SOFT_INPUT_ADJUST_PAN)
+
+fun Fragment.useAdjustResize() = setSoftInputMode(SOFT_INPUT_ADJUST_RESIZE)
+
+private fun Fragment.setSoftInputMode(flags: Int) {
+    (activity ?: return).window.setSoftInputMode(flags)
```
Comment (bychkovdmitry) [inline stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/util/extensions/Fragment.kt:15]: Maybe activity?.window?.setSoftInputMode(flags)?

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt [hunk 3]
```diff
@@ -56,6 +57,11 @@ class ChatFragment : Fragment() {
         configureBackButtonHandling()
     }
 
+    override fun onResume() {
+        super.onResume()
```
Comment (bychkovdmitry) [inline stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt:61]: Good, I was worried that client would use incorrect resize mode which I defined in the manifest 👍

**Review by bychkovdmitry — APPROVED** [review]

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/channel/add/AddChannelFragment.kt (hunk 1 above)
Comment (zsmb13) [reply stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/channel/add/AddChannelFragment.kt:44]: The really sad part is that this kind of dynamic resizing doesn't work if you don't have that in the manifest. I tried for a while, but it seems to be required, super weird. ([Same conclusion in this SO answer](https://stackoverflow.com/a/47629013/4465208))

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/util/extensions/Fragment.kt (hunk 2 above)
Comment (zsmb13) [reply stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/util/extensions/Fragment.kt:15]: It's the same thing, this just has fewer `?` in it 😄

**Review by adasiewiczr — APPROVED** [review]

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/channel/add/AddChannelFragment.kt (hunk 1 above)
Comment (bychkovdmitry) [reply stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/channel/add/AddChannelFragment.kt:44]: Facepalm

**Pull request merged**